userspace/libpman/src/lifecycle.c

```diff
@@ -28,6 +29,43 @@ int pman_open_probe() {
 	return 0;
 }
 
+int pman_prepare_progs_before_loading() {
+	/*
+	 * Probe required features for each bpf program, as requested
+	 */
+	for(int ev = 0; ev < PPM_EVENT_MAX; ev++) {
+		int num_skipped = 0;
+		int idx = 0;
+		event_prog_t *progs = event_prog_table[ev];
+		for(; idx < MAX_FEATURE_CHECKS && progs[idx].name != NULL; idx++) {
+			if(progs[idx].feat > 0) {
+				if(libbpf_probe_bpf_helper(BPF_PROG_TYPE_RAW_TRACEPOINT, progs[idx].feat, NULL) ==
+				   0) {
+					// Required feature not present
+					struct bpf_program *p =
+					        bpf_object__find_program_by_name(g_state.skel->obj, progs[idx].name);
+					if(p) {
+						bpf_program__set_autoload(p, false);
+					} else {
+						pman_print_error(" unable to find prog");
+					}
+					num_skipped++;
+				}
+			}
+		}
+		if(num_skipped > 0) {
+			if(num_skipped == idx) {
+				pman_print_error(" no program satisfies required features for event");
```

Comment:
Is there any way we could print _which_ event is not being satisfied? Even the index could make debugging a lot easier.

Reply:
A snprintf would do the trick! Will add it!